Adding storage gloss terms (#335)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/glossary_terms_conventions/general_conventions/d.adoc

*See also*: xref:debug-adj[debug (adjective)]

[[deduplication]]
==== image:images/yes.png[yes] deduplication (noun)

> *Description*:_Deduplication_ is a feature of virtual disk optimization (VDO) that helps conserve storage space by eliminating multiple copies of duplicate blocks. Instead of writing the same data more than once, VDO detects blocks that are duplicates of an existing block, and records them as a reference to the original block. VDO maintains a mapping from logical block addresses, which are used by the storage layer above VDO, to physical block addresses, which are used by the storage layer under VDO. After deduplication, multiple logical block addresses can be mapped to the same physical block address; these are called shared blocks. Block sharing is invisible to users of the storage, who read and write blocks as they would if VDO were not present. When a shared block is overwritten, a new physical block is allocated for storing the new block data to ensure that other logical block addresses that are mapped to the shared physical block are not modified. Use _block_ or _VDO_ or _local storage device_ instead of using the vague term _device_.

Reviewer: This is a very long glossary entry. Can it be trimmed down?

*See also*: xref:debug-adj[debug (adjective)]

[[deduplication]]
==== image:images/yes.png[yes] deduplication (noun)

> *Description*: A feature of virtual disk optimization (VDO) that helps conserve storage space by eliminating multiple copies of duplicate blocks. Instead of writing the same data more than once, VDO detects blocks that are duplicates of an existing block, and records them as a reference to the original block. VDO maintains a mapping from logical block addresses, which are used by the storage layer above VDO, to physical block addresses, which are used by the storage layer under VDO. After deduplication, multiple logical block addresses can be mapped to the same physical block address; these are called shared blocks. Block sharing is invisible to users of the storage, who read and write blocks as they would if VDO were not present. When a shared block is overwritten, a new physical block is allocated for storing the new block data to ensure that other logical block addresses that are mapped to the shared physical block are not modified. Use "block" or "VDO" or "local storage device" instead of using the vague term "device".

Reviewer 2: ```suggestion
*Description*: _Deduplication_ is a feature of virtual disk optimization (VDO) that helps conserve storage space by eliminating multiple copies of duplicate blocks. Instead of writing the same data more than once, VDO detects blocks that are duplicates of an existing block, and records them as a reference to the original block. VDO maintains a mapping from logical block addresses, which are used by the storage layer above VDO, to physical block addresses, which are used by the storage layer under VDO. After deduplication, multiple logical block addresses can be mapped to the same physical block address; these are called shared blocks. Block sharing is invisible to users of the storage, who read and write blocks as they would if VDO were not present. When a shared block is overwritten, a new physical block is allocated for storing the new block data to ensure that other logical block addresses that are mapped to the shared physical block are not modified. Use _block_ or _VDO_ or _local storage device_ instead of using the vague term _device_.
```
- Not sure who the `who` is in this sentence: `Block sharing is invisible to users of the storage, who read and write blocks as they would if VDO were not present`.
- On a related note, is this level of detail necessary here? It's much more detailed than other entries. Maybe consider cutting everything except the first and last sentences or finding some other method of paring this content down? Just a thought.

*See also*:

[[directory]]
==== image:images/yes.png[yes] directory (noun)

> *Description*: A _directory_ is a special type of file in a Linux file system that contains a list of objects and their inodes.

Reviewer 3: "inode" looks to be correct

*See also*:

[[directory]]
==== image:images/yes.png[yes] directory (noun)

> *Description*: A special type of file in a Linux file system that contains a list of objects and their inodes.

Reviewer 2: ```suggestion
*Description*: A _directory_ is a special type of file in a Linux file system that contains a list of objects and their inodes.
```

*See also*: xref:red-hat-directory-server[Red Hat Directory Server]

[[disaster-recovery]]
==== image:images/yes.png[yes] disaster recovery (adjective)

> *Description*: Related to processes or methods used to ensure data integrity and the operational continuity of an environment or infrastructure in case of malicious or accidental interruption.

Reviewer 2: ```suggestion
*Description*: _Disaster recovery_ refers to processes or methods used to ensure data integrity and the operational continuity of an environment or infrastructure in the event of malicious or accidental interruption.
```

*See also*:

[[disk-encryption]]
==== image:images/yes.png[yes] disk encryption (noun)

> *Description*:  _Disk encryption_ is a security feature to protect access to sensitive information stored on a disk by using mathematical functions to convert disk data into a format that cannot be deciphered easily by unauthorized processes or users. _Disk encryption_ is also known as _block device encryption_ or _encryption at rest_.

Reviewer: ```suggestion
*Description*:  _Disk encryption_ is a security feature to protect access to sensitive information stored on a disk by using mathematical functions to convert disk data into a format that cannot be deciphered easily by unauthorized processes or users. Disk encryption is also known as _block device encryption_ or _encryption at rest_.
```
Only the first usage is italicized.

*See also*:

[[disk-encryption]]
==== image:images/yes.png[yes] disk encryption (noun)

> *Description*: The encryption of data while it is stored on a disk. Also known as block device encryption or encryption at rest.

Reviewer 2: ```suggestion
*Description*: _Disk encryption_ of data while it is stored on a disk. Also known as block device encryption or encryption at rest.
```
- Defining _disk encryption` as `encryption of data on a disk` isn't really defining the term. Maybe something like:
 
`_Disk encryption_ is a security feature to protect access to sensitive information stored on a disk by using mathematical functions to convert disk data into a format that cannot be deciphered easily by unauthorized processes or users.`

*See also*: xref:amq-interconnect[AMQ Interconnect]

[[dispersed-volume]]
==== image:images/yes.png[yes] dispersed volume (noun)

> *Description*: A Gluster volume that uses erasure coding to write data across three or more bricks to ensure that data remains available even when a certain number of bricks is not available.

Reviewer 2: ```suggestion
*Description*: A _dispersed volume_ is a Gluster volume that uses erasure coding to write data across three or more bricks to ensure that data remains available even when a certain number of bricks is not available.
```

*See also*:

[discrete]
[[disconnected-deployment]]

> ==== image:images/caution.png[with caution] disconnected deployment or installation (noun)

Author: I'm not sure about removing this item. This isn't about storage, so I'm not comfortable making command decisions about removing/ammending.

> ==== image:images/caution.png[with caution] disconnected deployment or installation (noun)

Author: Done

*See also*:

[[distributed-file-system]]
==== image:images/yes.png[yes] distributed file system (noun)

> *Description*: A file system that presents files from a number of different storage devices, potentially on many different machines and in many different locations, as a single interface to an end user or consuming service.

Reviewer 2: ```suggestion
*Description*: A _distributed file system_ is a file system that presents files from a number of different storage devices, potentially on many different machines and in many different locations, as a single interface to an end user or consuming service.
```

*Use it*: yes

*Incorrect forms*:

*See also*:

[[distributed-volume]]
==== image:images/yes.png[yes] distributed volume (noun)

> *Description*: A Gluster volume that distributes data across one or more bricks.

Reviewer 2: ```suggestion
*Description*: A _distributed volume_ is a Gluster volume that distributes data across one or more bricks.
```

*Use it*: yes

*Incorrect forms*:

*See also*:

[distributed-dispersed-volume]]
==== image:images/yes.png[yes] distributed-dispersed volume (noun)

> *Description*: A Gluster volume that distributes and erasure codes data across one or more bricks.

Reviewer 2: ```suggestion
*Description*: A _distributed-dispersed volume_ is a Gluster volume that distributes and erasure codes data across one or more bricks.
```
- Also this part of the sentence seems to be missing something, or there's a typo: `that distributes and erasure codes data across one or more bricks.`

*Use it*: yes

*Incorrect forms*:

*See also*:

[[distributed-replicated-volume]]

> ==== image:images/yes.png[yes] distributed-dispersed volume (noun)

Reviewer 2: ```suggestion
==== image:images/yes.png[yes] distributed-replicated volume (noun)
```
fix typo

> *Description*: A Gluster volume that distributes and replicates data across multiple bricks.

Reviewer 2: ```suggestion
*Description*: A _distributed-replicated volume_ is a Gluster volume that distributes and replicates data across multiple bricks.
```

*Incorrect forms*: DVD burner, burner

*See also*:

[[dynamically-provisioned-storage]]
==== image:images/caution.png[with caution] dynamically provisioned storage (noun)

> *Description*:  _Dynamically provisioned storage_ is a data repository that changes in size depending on the current demand of the application or the user. In OpenShift Container Platform and OpenShift Data Foundation, use _dynamic volume provisioning_ to refer to the related `StorageClass` configuration parameter.

Reviewer: ```suggestion
*Description*:  _Dynamically provisioned storage_ is a data repository that changes in size depending on the current demand of the application or the user. In OpenShift Container Platform and OpenShift Data Foundation, use "dynamic volume provisioning" to refer to the related `StorageClass` configuration parameter.
```
